htdocs/src/Oc/User/UserRepository.php

```diff
@@ -2,7 +2,172 @@
 
 namespace Oc\User;
 
+use Doctrine\DBAL\Connection;
+
+/**
+ * Class UserRepository
+ *
+ * @package Oc\User
+ * @author Nick Lubisch <oc@nicklubisch.de>
+ */
 class UserRepository
 {
+    /**
+     * Database table name that this repository maintains.
+     *
+     * @var string
+     */
+    const TABLE = 'user';
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * UserRepository constructor.
+     *
+     * @param Connection $connection
+     */
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * Fetches all users.
+     *
+     * @return UserEntity[]
+     */
+    public function fetchAll()
+    {
+        $statement = $this->connection->createQueryBuilder()
+            ->select('*')
+            ->from(self::TABLE)
+            ->execute();
+
+        $result = $statement->fetchAll();
+
+        if ($result === false) {
+            return [];
+        }
+
+        return $this->getEntityArrayFromDatabaseArray($result);
+    }
+
+    /**
+     * Fetches a user by its id.
+     *
+     * @param int $id
+     *
+     * @return null|UserEntity
+     */
+    public function fetchOneById($id)
+    {
+        $statement = $this->connection->createQueryBuilder()
+            ->select('*')
+            ->from(self::TABLE)
+            ->where('user_id = :id')
+            ->setParameter(':id', $id)
+            ->execute();
+
+        $result = $statement->fetch();
+
+        if ($result === false) {
+            return null;
+        }
+
+        $entity = new UserEntity();
+
+        return $entity->fromDatabaseArray($result);
+    }
+
+    /**
+     * Creates a user in the database.
+     *
+     * @param UserEntity $entity
+     *
+     * @return UserEntity
+     */
+    public function create(UserEntity $entity)
+    {
+        if (!$entity->isNew()) {
+            throw new InvalidArgumentException('The entity does already exist.');
```

Comment:
the entity already exists?

Reply:
i introduce some custom exceptions for this